content/en/docs/tasks/debug-application-cluster/audit.md

```diff
@@ -231,187 +235,187 @@ and in the logs to monitor the state of the auditing subsystem.
-        </rule>
-    </match>
-
-    <filter audit.**>
+1. [Install `fluentd`](https://docs.fluentd.org/v1.0/articles/quickstart#step-1:-installing-fluentd),
+   adding the plugins `fluent-plugin-forest` and `fluent-plugin-rewrite-tag-filter`, onto the each
+   host that runs the kube-apiserver.
+
+   {{< note >}}
+   The `fluent-plugin-forest` and `fluent-plugin-rewrite-tag-filter` are plugins for fluentd.
+   You can get details about plugin installation from
+   [fluentd plugin-management](https://docs.fluentd.org/v1.0/articles/plugin-management).
+   {{< /note >}}
+
+1. Create a config file for fluentd (for this example: `/etc/fluentd/config`):
+   ```
+   # This example assumes that fluentd runs on the same host as the kube-apiserver, perhaps
+   # as a sidecar container.
+   <source>
+       @type tail
+       # audit log path of kube-apiserver
+       path /var/log/kube-audit
+       pos_file /var/log/audit.pos
+       format json
+       time_key time
+       time_format %Y-%m-%dT%H:%M:%S.%N%z
+       tag audit
+   </source>
+
+   <filter audit>
+       #https://github.com/fluent/fluent-plugin-rewrite-tag-filter/issues/13
        @type record_transformer
-       remove_keys namespace
-    </filter>
-
-    <match audit.**>
-        @type forest
-        subtype file
-        remove_prefix audit
-        <template>
-            time_slice_format %Y%m%d%H
-            compress gz
-            path /var/log/audit-${tag}.*.log
-            format json
-            include_time_key true
-        </template>
-    </match>
-    EOF
-    ```
+       enable_ruby
+       <record>
+        namespace ${record["objectRef"].nil? ? "none":(record["objectRef"]["namespace"].nil? ? "none":record["objectRef"]["namespace"])}
+       </record>
+   </filter>
+
+   <match audit>
+       # route audit according to namespace element in context
+       @type rewrite_tag_filter
+       <rule>
+           key namespace
+           pattern /^(.+)/
+           tag ${tag}.$1
+       </rule>
+   </match>
+
+   <filter audit.**>
+      @type record_transformer
+      remove_keys namespace
+   </filter>
+
+   <match audit.**>
+       @type forest
+       subtype file
+       remove_prefix audit
+       <template>
+           time_slice_format %Y%m%d%H
+           compress gz
+           path /var/log/audit-${tag}.*.log
+           format json
+           include_time_key true
+       </template>
+   </match>
+   ```
 
 1. Start fluentd
 
-    ```shell
-    fluentd -c /etc/fluentd/config  -vv
-    ```
+   ```shell
+   fluentd -c /etc/fluentd/config -vv
+   ```
 
-1. Start kube-apiserver with the following options:
+1. Start `kube-apiserver` with the following options:
 
-    ```shell
-    --audit-policy-file=/etc/kubernetes/audit-policy.yaml --audit-log-path=/var/log/kube-audit --audit-log-format=json
-    ```
+   ```shell
+   kube-apiserver ... --audit-policy-file=/etc/kubernetes/audit-policy.yaml --audit-log-path=/var/log/kube-audit --audit-log-format=json
```

Comment:
The `...` seems unclear. The original example would work if each flag is listed on a new line.

Reply:
_noted_